controller/api/models.py

```diff
@@ -298,6 +298,24 @@ def _start_containers(self, to_add):
         if set([c.state for c in to_add]) != set(['up']):
             err = 'warning, some containers failed to start'
             log_event(self, err, logging.WARNING)
+        # if the user specified a health check, try checking to see if it's running
+        try:
```

Comment:
Good call. Turns out that the proxy will respond and result in an aborted request, so I guess this would have to be re-written to keep trying for 60 seconds until the connection is not reset.

```
docker@boot2docker:~$ docker run -dp 8080:8080 deis/example-go bash -c 'sleep 60 && /go/bin/example-go'
6a24ec14c20663828c155a1e08d591b3e12d7e5d072782b78f31c7d0a8365d2e
docker@boot2docker:~$ docker run -v $PWD:/app -e HOST=192.168.59.104 python:2.7 bash -c 'pip install requests && python /app/test.py'
Collecting requests
  Downloading requests-2.7.0-py2.py3-none-any.whl (470kB)
Installing collected packages: requests
Successfully installed requests-2.7.0
Traceback (most recent call last):
  File "/app/test.py", line 4, in <module>
    response = requests.get('http://{}:8080'.format(os.environ['HOST']), timeout=60)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 69, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 50, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', error(104, 'Connection reset by peer'))
```

Script to test was as follows:

```
import os
import requests

response = requests.get('http://{}:8080'.format(os.environ['HOST']), timeout=60)

print response
```